src/main/java/edu/hm/hafner/analysis/SecureXmlParserFactory.java

```diff
@@ -81,6 +74,30 @@ public DocumentBuilder createDocumentBuilder() {
         }
     }
 
+    private void setFeature(final DocumentBuilderFactory factory, final String enabledProperty, final boolean status) {
+        try {
+            factory.setFeature(enabledProperty, status);
+        }
+        catch (ParserConfigurationException ignored) {
+            // ignore and continue
+        }
+    }
+
+    private void secureFactory(final DocumentBuilderFactory factory) {
+        try {
+            factory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+        }
+        catch (IllegalArgumentException e) {
```

Comment:
It seems that this attribute is not supported for the SAX parsers. Do you have an example where it is actually valid? In https://cheatsheetseries.owasp.org/cheatsheets/XML_External_Entity_Prevention_Cheat_Sheet.html it is also not referenced. Do you know why SonarQube does mention it but OWASP not? 

Should we rather use it for `XMLInputFactory`?